File: LeetCodes/add-digits/4-26-2023, 9_40_30 AM/add-digits.js
// Link: https://leetcode.com/problems/add-digits

/** Submission Info:
 *
 *  AUTHOR: the_247 | Raj Patel
 *  TITLE: add-digits
 *  RUNTIME: 69 ms
 *  MEMORY: 44.1 MB
 *  DATE: 4-26-2023, 9:40:30 AM
 *
 */

// Solution:

/**
 * @param {number} num
 * @return {number}
 */
var addDigits = function(num) {
    return 1 + (num - 1) % 9;
};

File: LeetCodes/add-digits/4-26-2023, 9_29_06 AM/add-digits.cpp
// Link: https://leetcode.com/problems/add-digits

/** Submission Info:
 *
 *  AUTHOR: the_247 | Raj Patel
 *  TITLE: add-digits
 *  RUNTIME: 0 ms
 *  MEMORY: 6 MB
 *  DATE: 4-26-2023, 9:29:06 AM
 *
 */

// Solution:

class Solution {
public:
    int addDigits(int num) {
        int n = num;
        while(n > 9) {
            int sum = 0;
            while(n > 0) {
                sum += n % 10;
                n /= 10;
            }
            n = sum;
        }
        return n;
    }
};
